llvm/docs/TableGen/ProgRef.rst

```diff
@@ -1646,7 +1646,8 @@ and non-0 as true.
 ``!and(``\ *a*\ ``,`` *b*\ ``, ...)``
```

Comment:
It seems this document is misleading, in the sense the !and/!or are binary operators in TableGen, but the syntax suggests that it can be > 2 operands. Something we need to fix (in a separate PR)?

Reply:
we can have more than 2 operands for both !and & !or. They're turned in to something like `!and(!and(a, b), c)` underlying and implemented with BinOpInit.

Reply:
Right, I see that now in TGParser. Thanks for clarifying.